.github/workflows/ledger-suite-release.yml

```diff
@@ -0,0 +1,657 @@
+          TEMP_DIR_METADATA=$(mktemp -d)
+          echo "Created temporary directory for metadata: $TEMP_DIR_METADATA"
+          echo "temp_dir_metadata=$TEMP_DIR_METADATA" >> $GITHUB_OUTPUT
+          # Store git log and command in files to handle multiline output
+          echo "$GIT_LOG" > "$TEMP_DIR_METADATA/git_log.txt"
+          echo "$GIT_COMMAND" > "$TEMP_DIR_METADATA/git_command.txt"
+          echo "git_log_file=$TEMP_DIR_METADATA/git_log.txt" >> $GITHUB_OUTPUT
+          echo "git_command_file=$TEMP_DIR_METADATA/git_command.txt" >> $GITHUB_OUTPUT
+
+      - name: Create release body
+        id: release_body
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+
+            const tempDirMetadata = '${{ steps.git_log.outputs.temp_dir_metadata }}';
+            const releaseType = '${{ steps.parse_tag.outputs.release_type }}';
+            const humanTitle = '${{ steps.variables.outputs.human_title }}';
+            const tagName = '${{ steps.parse_tag.outputs.tag_name }}';
+            const upgradeFrom = '${{ steps.variables.outputs.upgrade_from }}';
+            const repository = '${{ github.repository }}';
+            const tempDirArtifacts = '${{ steps.download.outputs.temp_dir_artifacts }}';
+
+            // Parse WASM files from JSON array
+            const wasmFiles = JSON.parse('${{ steps.variables.outputs.wasm_files }}');
+
+            // Define the release body template
+            const releaseTemplate = [
+              '### {{humanTitle}}',
+              '',
+              '#### Features',
+              '**⚠️ TODO: Replace this section with actual feature descriptions before publishing! ⚠️**',
+              '',
+              '#### Upgrade and Downgrade Notes',
+              '{{upgradeNotes}}- It is recommended to have all ledger suite canisters running at the same version.',
+              '- Upgrading the ledger suite canisters shall be done in the following order:',
+              '  1. The index first.',
+              '  2. The ledger second.',
+              '  3. Any archives third.',
+              '',
+              '#### WASM Verification',
+              '```',
+              'git fetch',
+              'git checkout {{tagName}}',
+              './ci/container/build-ic.sh -c',
+              '{{wasmCommands}}```',
+              'The hashes should match the values included in the `SHA256SUMS` file. For convenience, the filenames and corresponding hashes are also included in the table below.',
+              '',
+              '#### Change Log',
+              '',
+              '```',
+              '{{gitCommand}}{{gitLog}}',
+              '```',
+              '',
+              '#### Files and Hashes',
+              '',
+              '| Filename  | SHA256  |',
+              '| --- | --- |',
+              '{{hashTable}}'
+            ].join('\n');
```

Comment:
You could consider rewriting this more clearly by using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) which support multi-line strings and string-interpolation so you don't need the `releaseTemplate.replace` calls below.